### ISSUE TITLE
Red screen & unable to unlock when changing screen

### ISSUE THREAD

**Issue by jvanbruegge** [swaywm/swaylock#282]
This is with ext-session-lock (swaylock 1.7) and sway 1.8. If I lock my laptop, suspend it by closing it, then connect it to my docking station with two external monitors and resuming it, I get greeted by completely red screens on both monitors and no way to unlock except rebooting the machine.

I can see this in the logs:
```
[main.c:1188] Parsing CLI Args
[main.c:489] Loaded image <path> for output *
[main.c:1245] Using ext-session-lock-v1

<other logs>

ext_session_lock_surface_v1@33: error 2: committed surface dimensions do not match last acked configure
```

Not sure if that is related, but shortly after, waybar dumps core.

**Comment by kennylevinsen** [comment]
This means that swaylock hit a protocol error, stemming from this check: https://gitlab.freedesktop.org/wlroots/wlroots/-/blob/master/types/wlr_session_lock_v1.c#L145

If you can reproduce with swaylock 1.7.2 (or master), run swaylock manually with `WAYLAND_DEBUG=1` set in the environment, collecting output to a file as it crashes. This should show what causes the protocol error..

**Comment by jacksgt** [comment]
I am facing the exact same issue and started debugging, but then I realized that I'm in fact using `swaylock-effects` (1.6.10) instead of the regular `swaylock`.
With vanilla `swaylock` (1.7) screen locking works without issues on `sway` 1.8.

I see that there is another discussion for swaylock-effects: https://github.com/mortie/swaylock-effects/issues/97

**Comment by jvanbruegge** [comment]
Yes, it is really hard for me to reproduce and I am not even sure if 1.7.2 might have fixed it, can't remember if the last red screen was before or after the upgrade

**Comment by jvanbruegge** [comment]
Ok, I just had this today again, so it is indeed still an issue. Sadly I had no time for investigation so I have to wait until the next time this happens

**Comment by fberg** [comment]
This happens fairly frequently to me, and I captured logs with `WAYLAND_DEBUG=1`. Happened under similar circumstances as in the OP (suspend with internal display, wakeup with external monitors attached). I'm on swaylock 1.7.2.

Btw: you can unlock the session by switching to a different TTY and starting swaylock there (with appropriate `WAYLAND_DISPLAY` set).

[swaylock.error.log](https://github.com/swaywm/swaylock/files/12567891/swaylock.error.log)

**Comment by kennylevinsen** [comment]
> With sporadic bugs like this, I'd recommend always recording output with WAYLAND_DEBUG=1

Be very careful with this as such log also includes key codes of keypresses made during the execution of the application.

**Comment by kennylevinsen** [comment]
> I use waylock temporarily for now. Doesn't seem to have the issue. Although it's not as intuitive as swaylock-effects. But I prefer bug free functionality over visual form.

This is not the swaylock-effects project.

**Comment by Trust2k** [comment]
Initial configuration: The laptop is connected to an external monitor.

To reproduce the appearance of the red screen, just follow these steps::

1. Transfer the PC to lock or suspend.
2. Turn off the external monitor by pressing the button.
3. Make sure that the laptop is working.
4. Turn on the external monitor by pressing the button.
5. We get a red screen.

It plays stably.

**Comment by kennylevinsen** [comment]
> Whatever bugs swaylock has, all of its forks carry them too

No, forks deviate so they are not representative of upstream. They may inherit some bugs, fix other bugs and gain new bugs.

We don't care about promotion (we encourage alternatives), we just don't want issue reports or observations related to forks we do not work on and can do nothing about.

**Comment by ninetailedtori** [comment]
I CR this bug on my machine too, using swaylock latest git build :) seems to occur after a stable suspend duration of around 30min or more for me.

**Comment by Trust2k** [comment]
> I CR this bug on my machine too, using swaylock latest git build :) seems to occur after a stable suspend duration of around 30min or more for me.

the reason is a conflict with waybar
